packages/react/src/helpers.ts

```diff
@@ -58,10 +58,14 @@ export type PaginatedReadResult<T> = ReadResult<T> & {
   next: () => Promise<void>;
 };
 
+type PaginatedQueryResult<K> = {
+  result: { pageInfo: CommonPaginatedResultInfoFragment; items: K };
+};
+
 export function usePaginatedReadResult<
   K,
-  T extends { result: { pageInfo: CommonPaginatedResultInfoFragment; items: K } },
   V,
+  T extends PaginatedQueryResult<K> = PaginatedQueryResult<K>,
```

Comment:
Default value was needed for a previous attempt. But find this increased readability so I kept `PaginatedQueryResult` and its usage.

Reply:
Do we overall need that third generic parameter? Any benefits of keeping it over just inlining?

Reply:
I guess we technically don't... without will make the argument type slightly heavy to read (not strong about it)...

```ts
{ error, data, loading, fetchMore }: QueryResult<PaginatedQueryResult<K>, V>
```
Does this make any difference?

```ts
export function usePaginatedReadResult<
  K,
  V
  T = PaginatedQueryResult<K>
>(...) {...}
```

Reply:
no difference from what i can tell, just was curious why it's there.
personally, I just try to keep the amount of generic to what is really required as it makes the `hover over popup` easier to consume but not big deal.

Reply:
I can relate to it. Addressed here: https://github.com/lens-protocol/lens-sdk/pull/61/commits/7c92d07195fc6f8be041351d0bf7c0785cc93d3b

Reply:
@desfero I couldn't pin point why but if the result shape is not part of the generic TS struggles to narrow down the response of a generate query in these circumstances:
```ts
export function useSearchPublications({
  query,
  limit = DEFAULT_PAGINATED_QUERY_LIMIT,
  observerId,
}: UseSearchPublicationsArgs): PaginatedReadResult<(PostFragment | CommentFragment)[]> {
  const { sources, apolloClient } = useSharedDependencies();

  return usePaginatedReadResult(
    useSearchPublicationsQuery({
      variables: {
        query,
        sources,
        limit,
        observerId,
      },
      client: apolloClient,
    }),
  );
}

export function useSearchProfiles({
  query,
  limit = DEFAULT_PAGINATED_QUERY_LIMIT,
  observerId,
}: UseSearchProfilesArgs): PaginatedReadResult<ProfileFieldsFragment[]> {
  const { apolloClient } = useSharedDependencies();

  return usePaginatedReadResult(
    useSearchProfilesQuery({
      variables: {
        query,
        limit,
        observerId,
      },
      client: apolloClient,
    }),
  );
}
```

I don't think it's the reason why it was in the generic in first instance... just found out after rebasing my changes on top of recent `main`.